control/autoware_autonomous_emergency_braking/src/node.cpp

```diff
@@ -577,6 +610,93 @@ std::vector<Polygon2d> AEB::generatePathFootprint(
+    return autoware_universe_utils::createPoint(p.x, p.y, p.z);
+  }();
+
+  auto get_object_tangent_velocity =
+    [&](const PredictedObject & predicted_object, const auto & obj_pose) {
+      const double obj_vel_norm = std::hypot(
+        predicted_object.kinematics.initial_twist_with_covariance.twist.linear.x,
+        predicted_object.kinematics.initial_twist_with_covariance.twist.linear.y);
+
+      const auto obj_yaw = tf2::getYaw(obj_pose.orientation);
+      const auto obj_idx = motion_utils::findNearestIndex(ego_path, obj_pose.position);
+      const auto path_yaw = (current_ego_speed > 0.0)
+                              ? tf2::getYaw(ego_path.at(obj_idx).orientation)
+                              : tf2::getYaw(ego_path.at(obj_idx).orientation) + M_PI;
+      return obj_vel_norm * std::cos(obj_yaw - path_yaw);
+    };
+
+  geometry_msgs::msg::TransformStamped transform_stamped{};
+  try {
+    transform_stamped = tf_buffer_.lookupTransform(
+      "base_link", predicted_objects_ptr_->header.frame_id, stamp,
+      rclcpp::Duration::from_seconds(0.5));
+  } catch (tf2::TransformException & ex) {
+    RCLCPP_ERROR_STREAM(get_logger(), "[AEB] Failed to look up transform from base_link to map");
+    return;
+  }
+
+  // Check which objects collide with the ego footprints
+  std::for_each(objects.begin(), objects.end(), [&](const auto & predicted_object) {
+    // get objects in base_link frame
+    const auto t_predicted_object =
+      utils::transformObjectFrame(predicted_object, transform_stamped);
+    const auto & obj_pose = t_predicted_object.kinematics.initial_pose_with_covariance.pose;
+    const auto obj_poly = convertObjToPolygon(t_predicted_object);
+    const double obj_tangent_velocity = get_object_tangent_velocity(t_predicted_object, obj_pose);
+
+    for (const auto & ego_poly : ego_polys) {
+      // check collision with 2d polygon
+      std::vector<Point2d> collision_points_bg;
+      bg::intersection(ego_poly, obj_poly, collision_points_bg);
+      if (collision_points_bg.empty()) continue;
+
+      // Create an object for each intersection point
+      bool collision_points_added{false};
+      for (const auto & collision_point : collision_points_bg) {
+        const auto obj_position =
+          autoware_universe_utils::createPoint(collision_point.x(), collision_point.y(), 0.0);
+        const double obj_arc_length =
+          motion_utils::calcSignedArcLength(ego_path, current_p, obj_position);
+        if (std::isnan(obj_arc_length)) continue;
+
+        // If the object is behind the ego, we need to use the backward long offset. The
+        // distance should be a positive number in any case
+        const bool is_object_in_front_of_ego = obj_arc_length > 0.0;
+        const double dist_ego_to_object =
+          (is_object_in_front_of_ego) ? obj_arc_length - vehicle_info_.max_longitudinal_offset_m
+                                      : obj_arc_length + vehicle_info_.min_longitudinal_offset_m;
+
```

Comment:
Do you think we should also consider the bounding box size ?

Reply:
@soblin Hmm, the point used to calculate the arc length is the intersection between the ego footprint poly and the bounding box, I think that should be the distance. Do you have other suggestion?

Reply:
I see, that's right !